src/platforms/android/androidutils.cpp

```diff
@@ -134,3 +135,71 @@ void AndroidUtils::appReviewRequested() {
       "org/mozilla/firefox/vpn/qt/VPNAppReview", "appReviewRequested",
       "(Landroid/app/Activity;)V", activity.object<jobject>());
 }
+
+// static
+void AndroidUtils::dispatchToMainThread(std::function<void()> callback) {
+  QTimer* timer = new QTimer();
+  timer->moveToThread(qApp->thread());
+  timer->setSingleShot(true);
+  QObject::connect(timer, &QTimer::timeout, [=]() {
+    callback();
+    timer->deleteLater();
+  });
+  QMetaObject::invokeMethod(timer, "start", Qt::QueuedConnection);
+}
+
+// static
+QByteArray AndroidUtils::getQByteArrayFromJString(JNIEnv* env, jstring data) {
```

Comment:
@bakulf In a now outdated comment you had asked if "raw" contained anything sensitive. 

I've now moved all those functions into util helper functions. How should I handle debug logging as the utils could be used for sensitive data in the future?

Reply:
We should not log "data" in these methods. What you are doing here is perfect.